modules/social_features/social_event/social_event.module

```diff
@@ -519,6 +548,61 @@ function social_event_form_alter(&$form, FormStateInterface $form_state, $form_i
   }
 }
 
+/**
+ * Determine if we need to show a message with pending requests on the page.
+ */
+function social_event_show_pending_requests_message() {
+  $message = NULL;
+  // First lets check if the user has the correct permission.
+  if (\Drupal::currentUser()->hasPermission('manage event enrollment requests')) {
+    // Then get the current route to check on.
+    $current_route = \Drupal::routeMatch()->getRouteName();
+    // Defined routes on which a message with amount
+    // of unattended pending enrollment requests.
+    $defined_routes = [
+      'view.event_enrollments.view_enrollments',
+      'entity.node.canonical',
+      'view.event_manage_enrollments.page_manage_enrollments',
+      'view.managers.view_managers',
+      'social_event_managers.add_enrollees',
+      'entity.event_enrollment.delete_form',
+      'social_event_managers.vbo.execute_configurable',
+      'social_event_managers.vbo.confirm',
+    ];
+    // If the route matches, proceed.
+    if (in_array($current_route, $defined_routes)) {
+      // Third, get the current node (if any) to perform an extra check
+      // just to make sure that we are actually on an event node.
+      $node = social_event_get_current_event();
+      if ($node && $node->field_enroll_method->value === '2') {
+        // At this point we try to get the amount of pending requests.
+        try {
+          $query = \Drupal::entityTypeManager()
+            ->getStorage('event_enrollment')
+            ->getQuery();
+          $condition_group = $query->andConditionGroup()
+            ->condition('field_event.target_id', $node->id())
+            ->condition('field_request_status.value', 'pending')
+            ->condition('field_enrollment_status.value', '0');
+          $query->condition($condition_group);
+          $pending_requests = $query->execute();
+          if (!empty($pending_requests)) {
+            $link = Link::createFromRoute(t('(@amount) new requests', ['@amount' => count($pending_requests)]), 'view.event_manage_enrollment_requests.page_manage_enrollment_requests', ['node' => $node->id()])->toString();
```

Comment:
I know this might be tricky here, but it would be really nice if we can do the plural string translation here.

E.g. "There is 1 new request" and "There are 15 new requests".

You could use the `@amount` to create the plural string translations for both line `590` and `592`.

Reply:
Ah good one! I'll check it out, thanks 😄